### PR TITLE
Instant sealing for dev

### LINKED ISSUE #96: Support manual sealing
https://github.com/darwinia-network/darwinia-2.0/pull/68#discussion_r1040371065
Refer to Acala

### PR DESCRIPTION
**Pull request by jiguantong**
Close #96 

Refer to: 
https://github.com/AcalaNetwork/Acala/pull/2309
https://github.com/AcalaNetwork/Acala/pull/1438

https://github.com/paritytech/substrate/issues/8498

### CODE REVIEW

**Comment by jiguantong** [comment]
```
root@Debian:/home/workspace/darwinia-git/darwinia-2.0# ./target/release/darwinia --chain crab-dev --alice --base-path /tmp/darwinia-data/
2022-12-14 16:49:07 Darwinia
2022-12-14 16:49:07 ✌️  version 6.0.0-63ae56a8a4f
2022-12-14 16:49:07 ❤️  by Darwinia Network <hello@darwinia.network>, 2020-2022
2022-12-14 16:49:07 📋 Chain specification: Crab2 Development
2022-12-14 16:49:07 🏷  Node name: Alice
2022-12-14 16:49:07 👤 Role: AUTHORITY
2022-12-14 16:49:07 💾 Database: RocksDb at /tmp/darwinia-data/chains/crab2-development/db/full
2022-12-14 16:49:07 ⛓  Native runtime: Crab2-6000 (DarwiniaOfficialRust-0.tx0.au0)
2022-12-14 16:49:08 [pallet::staking] assembling new collators for new session 0 at #0
2022-12-14 16:49:08 [pallet::staking] assembling new collators for new session 1 at #0
2022-12-14 16:49:08 Parachain id: Id(2105)
2022-12-14 16:49:08 Parachain Account: 5Ec4AhNxga1JYLioRBNxfRnovheDELVbZTRSnKMgvSVPvNcN
2022-12-14 16:49:08 Parachain genesis state: 0x00000000000000000000000000000000000000000000000000000000000000000068c3bafa307a167b030faa5bff6d0df97266cc0cd6d3261ba211ae7c56941b0303170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c11131400
2022-12-14 16:49:08 Is collating: yes
2022-12-14 16:49:08 [pallet::staking] assembling new collators for new session 0 at #0
2022-12-14 16:49:08 [pallet::staking] assembling new collators for new session 1 at #0
2022-12-14 16:49:08 🔨 Initializing Genesis block/state (state: 0x4b37…b4bd, header-hash: 0x6dc1…e0d9)
2022-12-14 16:49:09 🔨 Running Frontier DB migration from version 1 to version 2. Please wait.
2022-12-14 16:49:09 ✔️ Successful Frontier DB migration from version 1 to version 2 (0 entries).
2022-12-14 16:49:09 🏷  Local node identity is: 12D3KooWRjYSCXKVk9P8iH8ENDPx7JaJLm7LeiuYGVJ3kxMWAzRS
2022-12-14 16:49:09 💻 Operating system: linux
2022-12-14 16:49:09 💻 CPU architecture: x86_64
2022-12-14 16:49:09 💻 Target environment: gnu
2022-12-14 16:49:09 💻 CPU: 12th Gen Intel(R) Core(TM) i9-12900K
2022-12-14 16:49:09 💻 CPU cores: 12
2022-12-14 16:49:09 💻 Memory: 15847MB
2022-12-14 16:49:09 💻 Kernel: 5.15.79.1-microsoft-standard-WSL2
2022-12-14 16:49:09 💻 Linux distribution: Debian GNU/Linux 11 (bullseye)
2022-12-14 16:49:09 💻 Virtual machine: yes
2022-12-14 16:49:09 📦 Highest known block at #0
2022-12-14 16:49:09 〽️ Prometheus exporter started at 127.0.0.1:9615
2022-12-14 16:49:09 Running JSON-RPC HTTP server: addr=127.0.0.1:9933, allowed origins=Some(["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"])
2022-12-14 16:49:09 Running JSON-RPC WS server: addr=127.0.0.1:9944, allowed origins=Some(["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"])
2022-12-14 16:49:12 🙌 Starting consensus session on top of parent 0x6dc1c7484af56c8e4a7d171096cb792bb3b49d70293ba0ee55cfae326a0ee0d9
2022-12-14 16:49:12 🎁 Prepared block for proposing at 1 (0 ms) [hash: 0xb825c4a749874910064152b5773e4f465549db6c9ed86bd55e30089a7d22ab54; parent_hash: 0x6dc1…e0d9; extrinsics (2): [0xa8cf…0175, 0xf9f9…2462]]
2022-12-14 16:49:12 🔖 Pre-sealed block for proposal at 1. Hash now 0xefd42751aef976d832f4af1d95ea3be63fe76041c857eba2e6523117fb4c1097, previously 0xb825c4a749874910064152b5773e4f465549db6c9ed86bd55e30089a7d22ab54.
2022-12-14 16:49:12 Possible safety violation: attempted to re-finalize last finalized block 0x6dc1c7484af56c8e4a7d171096cb792bb3b49d70293ba0ee55cfae326a0ee0d9
2022-12-14 16:49:12 ✨ Imported #1 (0xefd4…1097)
2022-12-14 16:49:14 💤 Idle (0 peers), best: #1 (0xefd4…1097), finalized #1 (0xefd4…1097), ⬇ 0 ⬆ 0
2022-12-14 16:49:19 💤 Idle (0 peers), best: #1 (0xefd4…1097), finalized #1 (0xefd4…1097), ⬇ 0 ⬆ 0
2022-12-14 16:49:24 🙌 Starting consensus session on top of parent 0xefd42751aef976d832f4af1d95ea3be63fe76041c857eba2e6523117fb4c1097
2022-12-14 16:49:24 🎁 Prepared block for proposing at 2 (1 ms) [hash: 0x635a16c093ae695794c93122ce2d19423b04caedb61c05959caf7ce7e033b2b7; parent_hash: 0xefd4…1097; extrinsics (2): [0x5e09…d7b3, 0x0713…43b9]]
2022-12-14 16:49:24 🔖 Pre-sealed block for proposal at 2. Hash now 0xd5f792becbd4ba7b2cad30bb0afb82f84723747fc368a98dd90db6057ff1901b, previously 0x635a16c093ae695794c93122ce2d19423b04caedb61c05959caf7ce7e033b2b7.
2022-12-14 16:49:24 Possible safety violation: attempted to re-finalize last finalized block 0xefd42751aef976d832f4af1d95ea3be63fe76041c857eba2e6523117fb4c1097
2022-12-14 16:49:24 ✨ Imported #2 (0xd5f7…901b)
2022-12-14 16:49:24 💤 Idle (0 peers), best: #2 (0xd5f7…901b), finalized #2 (0xd5f7…901b), ⬇ 0 ⬆ 0
2022-12-14 16:49:29 💤 Idle (0 peers), best: #2 (0xd5f7…901b), finalized #2 (0xd5f7…901b), ⬇ 0 ⬆ 0
2022-12-14 16:49:34 💤 Idle (0 peers), best: #2 (0xd5f7…901b), finalized #2 (0xd5f7…901b), ⬇ 0 ⬆ 0
2022-12-14 16:49:36 🙌 Starting consensus session on top of parent 0xd5f792becbd4ba7b2cad30bb0afb82f84723747fc368a98dd90db6057ff1901b
2022-12-14 16:49:36 🎁 Prepared block for proposing at 3 (1 ms) [hash: 0x894c091f263d1e96011ba25778b18f42ebc30a9fd4fcf5ba10b681613fbde222; parent_hash: 0xd5f7…901b; extrinsics (2): [0x4253…6f70, 0xdcb7…52e2]]
2022-12-14 16:49:36 🔖 Pre-sealed block for proposal at 3. Hash now 0x339ecd9e1968e1bdf2d689baef64800899da3643bce4439d5731038bb1ceb88c, previously 0x894c091f263d1e96011ba25778b18f42ebc30a9fd4fcf5ba10b681613fbde222.
2022-12-14 16:49:36 Possible safety violation: attempted to re-finalize last finalized block 0xd5f792becbd4ba7b2cad30bb0afb82f84723747fc368a98dd90db6057ff1901b
2022-12-14 16:49:36 ✨ Imported #3 (0x339e…b88c)
```

File: node/src/command.rs [hunk 1]
```diff
@@ -571,6 +568,23 @@ pub fn run() -> Result<()> {
 					if config.role.is_authority() { "yes" } else { "no" }
 				);
 
+				if chain_spec.is_dev() {
+					return if chain_spec.is_crab() {
```
Comment (AsceticBear) [inline node/src/command.rs:572]: Curious about the relationship between the `start_dev_node` and the `start_parachain_node` next, are they doing duplicated things?  Seems they all `start_network` at the end.

Comment (jiguantong) [reply node/src/command.rs:572]: They do part of the same thing, but `start_dev_node` can only be used for devnet, because it will be finalized instantly

**Review by hackfisher — APPROVED** [review]

**Comment by AurevoirXavier** [comment]
I'll review this later.

File: node/src/command.rs [hunk 2]
```diff
@@ -571,6 +568,23 @@ pub fn run() -> Result<()> {
...
+						crate::service::start_dev_node::<CrabRuntimeApi, CrabRuntimeExecutor>(config, &eth_rpc_config)
```
Comment (AurevoirXavier) [inline node/src/command.rs:573]: remove `crate::`

**Pull request merged**